### PR TITLE
feat(connections): Implement connection API with native query

### PR DESCRIPTION
**Pull request by jpukg**
## Issue ID
NA
## Summary
Implement connection API with native query
## Checklist

- [ ] Have you added the summary of what your changes do and why you'd like us to include them?
- [ ] Have you added the steps to implement this feature?
- [x] Have the files been linted and formatted?
- [ ] Have the docs been updated to match the changes in the PR?
- [ ] Have the tests been updated to match the changes in the PR?
- [ ] Have you run the tests locally to confirm they pass?

### How does this PR fix the problem, Screenshot Please?

image

### CODE REVIEW

File: src/main/java/com/linkedin/network/api/ConnectionController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	public ResponseEntity<ApiResponse> getAllUsers() {
+		ApiResponse apiResponse = new ApiResponse();
+		List<ProfileDTO> connections = connectionService.getConnections();
+		if (!CollectionUtils.isEmpty(connections)) {
+			apiResponse.setResponse(connections);
+			apiResponse.setStatus(HttpStatus.OK.toString());
+			apiResponse.setMessage("Found Connections");
+		} else {
+			apiResponse.setMessage("No active connections available");
+		}
```
Comment (ramsunvtech) [inline src/main/java/com/linkedin/network/api/ConnectionController.java:32]: @jpukg, can this be replaced without inverse condition?

```
if (CollectionUtils.isEmpty(connections)) {
    apiResponse.setStatus(HttpStatus.OK.toString()); // update status code
    apiResponse.setMessage("No active connections available");
} else {
    apiResponse.setResponse(connections);
    apiResponse.setStatus(HttpStatus.OK.toString());
    apiResponse.setMessage("Found Connections");
}
  return ResponseEntity.ok().body(apiResponse);
```

File: src/main/java/com/linkedin/network/service/impl/ConnectionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import com.linkedin.network.service.*;
+
+@Service
+public class ConnectionServiceImpl implements ConnectionService {
+
+	private final ProfileRepository profileRepository;
+
+	public ConnectionServiceImpl(ProfileRepository profileRepository) {
+		this.profileRepository = profileRepository;
+	}
```
Comment (ramsunvtech) [inline src/main/java/com/linkedin/network/service/impl/ConnectionServiceImpl.java:20]: can this be auto wired. i mean this look like boilerplate code. correct me here

**Review by ramsunvtech — CHANGES_REQUESTED** [review]

File: src/main/java/com/linkedin/network/api/ConnectionController.java (hunk 1 above)
Comment (jpukg) [reply src/main/java/com/linkedin/network/api/ConnectionController.java:32]: @ramsunvtech will be given as a task to you guys.

File: src/main/java/com/linkedin/network/service/impl/ConnectionServiceImpl.java (hunk 2 above)
Comment (jpukg) [reply src/main/java/com/linkedin/network/service/impl/ConnectionServiceImpl.java:20]: @ramsunvtech yes, this will be corrected by you guys.

**Comment by ramsunvtech** [comment]
@jpukg please share the working screenshot and DB tables and update the postman collection

File: src/main/java/com/linkedin/network/api/ConnectionController.java (hunk 1 above)
Comment (shangan23) [reply src/main/java/com/linkedin/network/api/ConnectionController.java:32]: what is the inverse condition?

**Comment by jpukg** [comment]
image

**Pull request closed without merging**